controllers/tests/common_test.go

```diff
@@ -104,14 +104,37 @@ func createMinimalKafkaClusterCR(name, namespace string) *v1beta1.KafkaCluster {
 }
 
 func waitForClusterRunningState(kafkaCluster *v1beta1.KafkaCluster, namespace string) {
-	Eventually(func() (v1beta1.ClusterState, error) {
-		createdKafkaCluster := &v1beta1.KafkaCluster{}
-		err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: kafkaCluster.Name, Namespace: namespace}, createdKafkaCluster)
-		if err != nil {
-			return v1beta1.KafkaClusterReconciling, err
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
This can check be simplified as follows:
```
Consistently(func() (v1beta1.ClusterState, error) {
  createdKafkaCluster := &v1beta1.KafkaCluster{}
  err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: kafkaCluster.Name, Namespace: namespace}, createdKafkaCluster)
		if err != nil {
			return "", err
        }

       return createdKafkaCluster.Status.State, nil
}, 5*time.Second, 100*time.Millisecond).Should(Equal(v1beta1.KafkaClusterRunning))
```